chall09/egarcia-.c

```diff
@@ -0,0 +1,179 @@
+#include <string.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <ctype.h>
+
+char	*ft_strjoin(char const *s1, char const *s2)
+{
+	char	*str;
+	int		i;
+
+	i = 0;
+	if (s1 && s2)
+	{
+		if (!(str = (char*)malloc((strlen(s1) + strlen(s2) + 1)
+			* sizeof(char))))
+			return (NULL);
+		while (*s1)
+			str[i++] = *s1++;
+		while (*s2)
+			str[i++] = *s2++;
+		str[i] = '\0';
+		return (str);
+	}
+	return (0);
+}
+
+char   *ft_print_format(int y ,int d, int h, int m, int s)
+{
+    char *str;
+    char *tmp;
+    char *years = strdup("");
+    char *days = strdup("");
+    char *hours = strdup("");
+    char *minutes = strdup("");
+    char *seconds = strdup("");
+    if ( y > 0)
+    {
+
+        sprintf(years, "%d", y);
+        tmp = y > 1 ? ft_strjoin(years, " years") : ft_strjoin(years, " year");
```

Comment:
So many allocations! Why don't you simply do:
```suggestion
        sprintf(years, "%d %s", y, (y > 1 ? "years" : "year"));
```
This commentary is the same for every other cases.